### PR TITLE
test(Stepper): component and useStepper tests are added

### PR DESCRIPTION
**Pull request by aktasmehmet**
## Description

Tests that cover ``Stepper`` component and ``useStepper`` companion has been added.

_NOTE: In order to keep things simple (since there are too many tests) I had to split the tests into 2 describe blocks._

## Type of Change

- [ ] 🆕 New Component
- [x] ✨ Feature / Enhancement
- [ ] 🐛 Bug Fix
- [ ] 💥 Breaking Change
- [ ] 📦 Build / Dependency / Docs Update
- [ ] 🧹 Code Refactoring

## Screenshots / Preview

Screenshot 2026-06-19 at 14 13 30
Screenshot 2026-06-19 at 14 13 12

## Checklist

- [x] Self-reviewed, no leftover logs or debug code
- [ ] Uses design tokens — no hardcoded style values
- [ ] Accessible (keyboard nav, ARIA, contrast)
- [x] Tests added/updated and passing
- [ ] Storybook story added/updated

## How to Test

Run jest

### CODE REVIEW

File: src/lib/components/Stepper/Stepper.test.tsx [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+    // itemOrientation = vertical
+    expect(getRoot()).toHaveClass("vertical-items");
+
+    // variant = primary
+    expect(container.querySelector(".active")).toHaveClass("primary");
+  });
+
+  it("should render in the orientation given in orientation prop", () => {
+    const { rerender, getRoot } = renderExt({ orientation: "horizontal" });
```
Comment (atakanbayrak) [inline src/lib/components/Stepper/Stepper.test.tsx:50]: Do we need to give horizontal prop specifically because it is already default value in here?

File: src/lib/components/Stepper/Stepper.test.tsx [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    (["primary", "secondary", "success", "warning", "danger"] as const).forEach(variant => {
+      const { container, unmount } = renderExt({ variant });
+      expect(container.querySelector(".active")).toHaveClass(variant);
+      unmount();
+    });
+  });
+
+  it("should render step counter when showCount prop is true", () => {
+    const { container, rerender } = renderExt();
+    expect(container.querySelector(".stepCount")).not.toBeInTheDocument();
```
Comment (atakanbayrak) [inline src/lib/components/Stepper/Stepper.test.tsx]: querySelector based container selections have least hierarchy in test coverage after test id's. Is there any better way to track and check this situation?

File: src/lib/components/Stepper/Stepper.test.tsx [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+    rerender(
+      <Stepper showCount>
+        <Stepper.Item title="Step 1">Content 1</Stepper.Item>
+      </Stepper>,
+    );
+    expect(container.querySelector(".stepCount")).toBeInTheDocument();
+  });
+
+  it("should render the correct initial active step using defaultActiveStep", () => {
```
Comment (atakanbayrak) [inline src/lib/components/Stepper/Stepper.test.tsx:98]: defaultActiveStep prop is number based so we could add two more tests which would check if there is negative number or number which is more than step number.

File: src/lib/components/Stepper/Stepper.test.tsx [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+    const { container } = renderExt({ defaultActiveStep: 1 });
+    const stepItems = container.querySelectorAll(".stepItem");
+    expect(stepItems[0]).toHaveClass("completed");
+    expect(stepItems[1]).toHaveClass("active");
+    expect(stepItems[2]).toHaveClass("upcoming");
+  });
+
+  it("should render step content of the active step only", () => {
```
Comment (atakanbayrak) [inline src/lib/components/Stepper/Stepper.test.tsx:115]: We could enhance this test to check mid content because of previous and next content appearance. Previous step should be disappeared. Next step is already checked in here.

**Review by atakanbayrak — CHANGES_REQUESTED** [review]

File: src/lib/components/Stepper/Stepper.test.tsx (hunk 1 above)
Comment (aktasmehmet) [reply src/lib/components/Stepper/Stepper.test.tsx:50]: yes necessary. since this is not a default value check, all props should be given. It is like checking size props of a button.

File: src/lib/components/Stepper/Stepper.test.tsx (hunk 2 above)
Comment (aktasmehmet) [reply src/lib/components/Stepper/Stepper.test.tsx]: Yes but a little complicated. Since there are 2 seperate spans there, it is not possible to check with a text. I am implementing to check the item with the right side of the text, after the slash.

File: src/lib/components/Stepper/Stepper.test.tsx (hunk 4 above)
Comment (aktasmehmet) [reply src/lib/components/Stepper/Stepper.test.tsx:115]: This is a topic of another test I think. This test should just check the content of the active step is rendered. The scenario you mentioned are covered in lots of other places. The first example comes to my mind is: "should navigate to previous step when the previous button is clicked". and there are more similar

File: src/lib/components/Stepper/Stepper.test.tsx (hunk 3 above)
Comment (aktasmehmet) [reply src/lib/components/Stepper/Stepper.test.tsx:98]: It was not checked before, thats the reason of the missing test. I'm implementing it. Upper bound in useStepper will not be able to be clamped because of the nature of that hook. It doesnt know the number of items. But in the component itself, it will work as expected.

**Review by atakanbayrak — APPROVED** [review]
